src/initial_conditions.jl

```diff
@@ -1,86 +1,114 @@
+abstract type ZonalWind <: InitialConditions end
+abstract type StartFromRest <: InitialConditions end
+abstract type StartFromFile <: InitialConditions end
+abstract type StartWithVorticity <: InitialConditions end
+
 """
     prognostic_variables = initial_conditions(M::ModelSetup)
 
 Initialize the prognostic variables from rest, with some initial vorticity,
 or restart from file."""
-function initial_conditions(M::ModelSetup)
-
-    progn = initialize_from_rest(M)             # allocate variables in any case
-    @unpack initial_conditions = M.parameters
-    @unpack radius_earth = M.geometry
+function initial_conditions(model::ModelSetup)
 
-    if initial_conditions == :rest
-        nothing 
+    progn = allocate_prognostic_variables(model)    # allocate variables in any case
+    IC = model.parameters.initial_conditions        # type of initial conditions
 
-    elseif initial_conditions == :barotropic_vorticity              # zonal wind with perturbation
+    initial_conditions!(IC,progn,model)     # dispatch to the type of initial conditions
 
-        mmax = min(15,progn.mmax+1)     # perturb only larger modes
-        lmax = min(15,progn.lmax+1)
+    # SCALING: we use vorticity*radius,divergence*radius in the dynamical core
```

Comment:
Scaling is applied to all initial conditions, this means that all input is assumed to be in SI units. I think that's fine, but it should be mentioned in the docstring

Reply:
Added a line to the docstring for that.